File: src/main/java/com/soapdogg/lispInterpreter/comparator/NodeValueComparator.java
package com.soapdogg.lispInterpreter.comparator;

import com.soapdogg.lispInterpreter.constants.ReservedValuesConstants;
import lombok.AllArgsConstructor;

@AllArgsConstructor(staticName = "newInstance")
public class NodeValueComparator {

    public boolean equalsNil(final String value){
        return ReservedValuesConstants.NIL.equals(value);
    }

    public boolean equalsT(final String value){
        return ReservedValuesConstants.T.equals(value);
    }
}

File: src/main/java/com/soapdogg/lispInterpreter/comparator/NodeValueComparator.kt
package com.soapdogg.lispInterpreter.comparator

import com.soapdogg.lispInterpreter.constants.ReservedValuesConstants

class NodeValueComparator {
    fun equalsNil(value: String): Boolean {
        return ReservedValuesConstants.NIL == value
    }

    fun equalsT(value: String): Boolean {
        return ReservedValuesConstants.T == value
    }
}